// sanity-plugin-mux-input/build/components/MuxLogo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = MuxLogo;
var _autoId = require("@reach/auto-id");
var _react = _interopRequireWildcard(require("react"));
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }
function _iterableToArrayLimit(arr, i) { var _i = arr == null ? null : typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"]; if (_i == null) return; var _arr = []; var _n = true; var _d = false; var _s, _e; try { for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
// Mapping ids to ensure filter refs are unique, otherwise all Logo instances are hidden if the first SVG is hidden, due to how SVGs deal with relative links and ids
var ids = ['title', 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r'];
function MuxLogo() {
  var id = (0, _autoId.useId)();
  var _useMemo = (0, _react.useMemo)(() => ids.map(field => "".concat(id, "-").concat(field)), [id]),
    _useMemo2 = _slicedToArray(_useMemo, 19),
    titleId = _useMemo2[0],
    a = _useMemo2[1],
    b = _useMemo2[2],
    c = _useMemo2[3],
    d = _useMemo2[4],
    e = _useMemo2[5],
    f = _useMemo2[6],
    g = _useMemo2[7],
    h = _useMemo2[8],
    i = _useMemo2[9],
    j = _useMemo2[10],
    k = _useMemo2[11],
    l = _useMemo2[12],
    m = _useMemo2[13],
    n = _useMemo2[14],
    o = _useMemo2[15],
    p = _useMemo2[16],
    q = _useMemo2[17],
    r = _useMemo2[18];
  return /*#__PURE__*/_react.default.createElement("svg", {
    "aria-labelledby": titleId,
    role: "img",
    xmlns: "http://www.w3.org/2000/svg",
    xmlSpace: "preserve",
    viewBox: "92.08878326416016 102.66712188720703 692.76123046875 219.99948120117188",
    style: {
      height: '26px'
    }
  }, /*#__PURE__*/_react.default.createElement("title", {
    id: titleId
  }, "Mux Logo"), /*#__PURE__*/_react.default.createElement("defs", null, /*#__PURE__*/_react.default.createElement("linearGradient", {
    id: c,
    spreadMethod: "pad",
    gradientTransform: "matrix(528.38055 0 0 -528.38055 63.801 159.5)",
    gradientUnits: "userSpaceOnUse",
    y2: 0,
    x2: 1,
    y1: 0,
    x1: 0
  }, /*#__PURE__*/_react.default.createElement("stop", {
    offset: 0,
    style: {
      stopOpacity: 1,
      stopColor: '#ff4e00'
    }
  }), /*#__PURE__*/_react.default.createElement("stop", {
    offset: 1,
    style: {
      stopOpacity: 1,
      stopColor: '#ff1791'
    }
  })), /*#__PURE__*/_react.default.createElement("linearGradient", {
    id: d,
    spreadMethod: "pad",
    gradientTransform: "matrix(523.66766 0 0 -523.66766 67.897 159.5)",
    gradientUnits: "userSpaceOnUse",
    y2: 0,
    x2: 1,
    y1: 0,
    x1: 0
  }, /*#__PURE__*/_react.default.createElement("stop", {
    offset: 0,
    style: {
      stopOpacity: 1,
      stopColor: '#ff4e00'
    }
  }), /*#__PURE__*/_react.default.createElement("stop", {
    offset: 1,
    style: {
      stopOpacity: 1,
      stopColor: '#ff1791'
    }
  })), /*#__PURE__*/_react.default.createElement("linearGradient", {
    id: g,
    spreadMethod: "pad",
    gradientTransform: "rotate(180 296.075 79.75) scale(524.84045)",
    gradientUnits: "userSpaceOnUse",
    y2: 0,
    x2: 1,
    y1: 0,
    x1: 0
  }, /*#__PURE__*/_react.default.createElement("stop", {
    offset: 0,
    style: {
      stopOpacity: 1,
      stopColor: '#ff4e00'
    }
  }), /*#__PURE__*/_react.default.createElement("stop", {
    offset: 1,
    style: {
      stopOpacity: 1,
      stopColor: '#ff1791'
    }
  })), /*#__PURE__*/_react.default.createElement("linearGradient", {
    id: i,
    spreadMethod: "pad",
    gradientTransform: "matrix(524.84045 0 0 -524.84045 63.801 159.5)",
    gradientUnits: "userSpaceOnUse",
    y2: 0,
    x2: 1,
    y1: 0,
    x1: 0
  }, /*#__PURE__*/_react.default.createElement("stop", {
    offset: 0,
    style: {
      stopOpacity: 1,
      stopColor: '#ff4e00'
    }
  }), /*#__PURE__*/_react.default.createElement("stop", {
    offset: 1,
    style: {
      stopOpacity: 1,
      stopColor: '#ff1791'
    }
  })), /*#__PURE__*/_react.default.createElement("linearGradient", {
    id: j,
    spreadMethod: "pad",
    gradientTransform: "matrix(523.08514 0 0 -523.08514 67.897 224.446)",
    gradientUnits: "userSpaceOnUse",
    y2: 0,
    x2: 1,
    y1: 0,
    x1: 0
  }, /*#__PURE__*/_react.default.createElement("stop", {
    offset: 0,
    style: {
      stopOpacity: 1,
      stopColor: '#ff4e00'
    }
  }), /*#__PURE__*/_react.default.createElement("stop", {
    offset: 1,
    style: {
      stopOpacity: 1,
      stopColor: '#ff1791'
    }
  })), /*#__PURE__*/_react.default.createElement("linearGradient", {
    id: k,
    spreadMethod: "pad",
    gradientTransform: "matrix(524.84045 0 0 -524.84045 63.801 94.553)",
    gradientUnits: "userSpaceOnUse",
    y2: 0,
    x2: 1,
    y1: 0,
    x1: 0
  }, /*#__PURE__*/_react.default.createElement("stop", {
    offset: 0,
    style: {
      stopOpacity: 1,
      stopColor: '#ff4e00'
    }
  }), /*#__PURE__*/_react.default.createElement("stop", {
    offset: 1,
    style: {
      stopOpacity: 1,
      stopColor: '#ff1791'
    }
  })), /*#__PURE__*/_react.default.createElement("linearGradient", {
    id: l,
    spreadMethod: "pad",
    gradientTransform: "matrix(524.84045 0 0 -524.84045 63.801 159.5)",
    gradientUnits: "userSpaceOnUse",
    y2: 0,
    x2: 1,
    y1: 0,
    x1: 0
  }, /*#__PURE__*/_react.default.createElement("stop", {
    offset: 0,
    style: {
      stopOpacity: 1,
      stopColor: '#ff4e00'
    }
  }), /*#__PURE__*/_react.default.createElement("stop", {
    offset: 1,
    style: {
      stopOpacity: 1,
      stopColor: '#ff1791'
    }
  })), /*#__PURE__*/_react.default.createElement("linearGradient", {
    id: m,
    spreadMethod: "pad",
    gradientTransform: "matrix(524.84045 0 0 -524.84045 63.801 94.554)",
    gradientUnits: "userSpaceOnUse",
    y2: 0,
    x2: 1,
    y1: 0,
    x1: 0
  }, /*#__PURE__*/_react.default.createElement("stop", {
    offset: 0,
    style: {
      stopOpacity: 1,
      stopColor: '#ff4e00'
    }
  }), /*#__PURE__*/_react.default.createElement("stop", {
    offset: 1,
    style: {
      stopOpacity: 1,
      stopColor: '#ff1791'
    }
  })), /*#__PURE__*/_react.default.createElement("linearGradient", {
    id: p,
    spreadMethod: "pad",
    gradientTransform: "matrix(521.97632 0 0 -521.97632 69.067 191.973)",
    gradientUnits: "userSpaceOnUse",
    y2: 0,
    x2: 1,
    y1: 0,
    x1: 0
  }, /*#__PURE__*/_react.default.createElement("stop", {
    offset: 0,
    style: {
      stopOpacity: 1,
      stopColor: '#ff4e00'
    }
  }), /*#__PURE__*/_react.default.createElement("stop", {
    offset: 1,
    style: {
      stopOpacity: 1,
      stopColor: '#ff1791'
    }
  })), /*#__PURE__*/_react.default.createElement("linearGradient", {
    id: q,
    spreadMethod: "pad",
    gradientTransform: "matrix(523.09039 0 0 -523.09039 67.312 191.973)",
    gradientUnits: "userSpaceOnUse",
    y2: 0,
    x2: 1,
    y1: 0,
    x1: 0
  }, /*#__PURE__*/_react.default.createElement("stop", {
    offset: 0,
    style: {
      stopOpacity: 1,
      stopColor: '#ff4e00'
    }
  }), /*#__PURE__*/_react.default.createElement("stop", {
    offset: 1,
    style: {
      stopOpacity: 1,
      stopColor: '#ff1791'
    }
  })), /*#__PURE__*/_react.default.createElement("linearGradient", {
    id: r,
    spreadMethod: "pad",
    gradientTransform: "matrix(524.84045 0 0 -524.84045 63.801 159.5)",
    gradientUnits: "userSpaceOnUse",
    y2: 0,
    x2: 1,
    y1: 0,
    x1: 0
  }, /*#__PURE__*/_react.default.createElement("stop", {
    offset: 0,
    style: {
      stopOpacity: 1,
      stopColor: '#ff4e00'
    }
  }), /*#__PURE__*/_react.default.createElement("stop", {
    offset: 1,
    style: {
      stopOpacity: 1,
      stopColor: '#ff1791'
    }
  })), /*#__PURE__*/_react.default.createElement("clipPath", {
    id: a,
    clipPathUnits: "userSpaceOnUse"
  }, /*#__PURE__*/_react.default.createElement("path", {
    d: "M0 319h657.706V0H0Z"
  })), /*#__PURE__*/_react.default.createElement("clipPath", {
    id: b,
    clipPathUnits: "userSpaceOnUse"
  }, /*#__PURE__*/_react.default.createElement("path", {
    d: "M423.64 242h164.999V77H423.64Z"
  })), /*#__PURE__*/_react.default.createElement("clipPath", {
    id: e,
    clipPathUnits: "userSpaceOnUse"
  }, /*#__PURE__*/_react.default.createElement("path", {
    d: "M0 319h657.706V0H0Z"
  })), /*#__PURE__*/_react.default.createElement("clipPath", {
    id: f,
    clipPathUnits: "userSpaceOnUse"
  }, /*#__PURE__*/_react.default.createElement("path", {
    d: "M311.3 242h93.031V77H311.3Z"
  })), /*#__PURE__*/_react.default.createElement("clipPath", {
    id: h,
    clipPathUnits: "userSpaceOnUse"
  }, /*#__PURE__*/_react.default.createElement("path", {
    d: "M198.96 242h35.106V77H198.96Z"
  })), /*#__PURE__*/_react.default.createElement("clipPath", {
    id: n,
    clipPathUnits: "userSpaceOnUse"
  }, /*#__PURE__*/_react.default.createElement("path", {
    d: "M0 319h657.706V0H0Z"
  })), /*#__PURE__*/_react.default.createElement("clipPath", {
    id: o,
    clipPathUnits: "userSpaceOnUse"
  }, /*#__PURE__*/_react.default.createElement("path", {
    d: "M69.067 242H169.12V141.947H69.067Z"
  }))), /*#__PURE__*/_react.default.createElement("g", {
    clipPath: "url(#".concat(a, ")"),
    transform: "matrix(1.33333 0 0 -1.33333 0 425.333)"
  }, /*#__PURE__*/_react.default.createElement("g", {
    style: {
      opacity: 0.69999701
    },
    clipPath: "url(#".concat(b, ")")
  }, /*#__PURE__*/_react.default.createElement("path", {
    style: {
      fill: "url(#".concat(c, ")"),
      stroke: 'none'
    },
    d: "M558.674 82.142c6.855-6.855 17.969-6.855 24.824 0 6.854 6.855 6.854 17.969 0 24.823L453.605 236.858c-6.855 6.855-17.969 6.855-24.824 0s-6.855-17.969 0-24.823z"
  }))), /*#__PURE__*/_react.default.createElement("path", {
    style: {
      fill: "url(#".concat(d, ")"),
      stroke: 'none'
    },
    d: "M558.674 236.858 428.781 106.966c-6.855-6.855-6.855-17.969 0-24.825 6.855-6.854 17.969-6.854 24.823 0l129.894 129.894c6.854 6.855 6.854 17.968 0 24.823A17.498 17.498 0 0 1 571.086 242a17.495 17.495 0 0 1-12.412-5.142",
    transform: "matrix(1.33333 0 0 -1.33333 0 425.333)"
  }), /*#__PURE__*/_react.default.createElement("g", {
    clipPath: "url(#".concat(e, ")"),
    transform: "matrix(1.33333 0 0 -1.33333 0 425.333)"
  }, /*#__PURE__*/_react.default.createElement("g", {
    style: {
      opacity: 0.69999701
    },
    clipPath: "url(#".concat(f, ")")
  }, /*#__PURE__*/_react.default.createElement("path", {
    style: {
      fill: "url(#".concat(g, ")"),
      stroke: 'none'
    },
    d: "M328.853 112.107c22.297 0 40.372 18.075 40.372 40.372v71.315c0 10.054 7.505 18.206 17.554 18.206 10.048 0 17.552-8.152 17.552-18.206v-71.315c0-41.686-33.793-75.479-75.478-75.479-9.694 0-17.553 7.859-17.553 17.554 0 9.694 7.859 17.553 17.553 17.553"
  })), /*#__PURE__*/_react.default.createElement("g", {
    style: {
      opacity: 0.69999701
    },
    clipPath: "url(#".concat(h, ")")
  }, /*#__PURE__*/_react.default.createElement("path", {
    style: {
      fill: "url(#".concat(i, ")"),
      stroke: 'none'
    },
    d: "M216.513 242c-10.049 0-17.553-8.152-17.553-18.206V95.206c0-10.054 7.504-18.206 17.553-18.206 10.048 0 17.553 8.152 17.553 18.206v128.588c0 10.054-7.505 18.206-17.553 18.206"
  }))), /*#__PURE__*/_react.default.createElement("path", {
    style: {
      fill: "url(#".concat(j, ")"),
      stroke: 'none'
    },
    d: "M369.225 224.447c0-9.694 7.859-17.553 17.553-17.553 9.695 0 17.553 7.859 17.553 17.553s-7.858 17.552-17.553 17.552c-9.694 0-17.553-7.858-17.553-17.552",
    transform: "matrix(1.33333 0 0 -1.33333 0 425.333)"
  }), /*#__PURE__*/_react.default.createElement("path", {
    style: {
      fill: "url(#".concat(k, ")"),
      stroke: 'none'
    },
    d: "M553.532 94.554c0-9.695 7.859-17.554 17.553-17.554 9.695 0 17.554 7.859 17.554 17.554 0 9.694-7.859 17.552-17.554 17.552-9.694 0-17.553-7.858-17.553-17.552",
    transform: "matrix(1.33333 0 0 -1.33333 0 425.333)"
  }), /*#__PURE__*/_react.default.createElement("path", {
    style: {
      fill: "url(#".concat(l, ")"),
      stroke: 'none'
    },
    d: "M69.067 223.794V95.206C69.067 85.152 76.571 77 86.62 77c10.048 0 17.553 8.152 17.553 18.206v128.588c0 10.055-7.505 18.205-17.553 18.205-10.049 0-17.553-8.15-17.553-18.205",
    transform: "matrix(1.33333 0 0 -1.33333 0 425.333)"
  }), /*#__PURE__*/_react.default.createElement("path", {
    style: {
      fill: "url(#".concat(m, ")"),
      stroke: 'none'
    },
    d: "M198.96 94.554c0-9.695 7.859-17.554 17.553-17.554 9.695 0 17.554 7.859 17.554 17.554 0 9.694-7.859 17.553-17.554 17.553-9.694 0-17.553-7.859-17.553-17.553",
    transform: "matrix(1.33333 0 0 -1.33333 0 425.333)"
  }), /*#__PURE__*/_react.default.createElement("g", {
    clipPath: "url(#".concat(n, ")"),
    transform: "matrix(1.33333 0 0 -1.33333 0 425.333)"
  }, /*#__PURE__*/_react.default.createElement("g", {
    style: {
      opacity: 0.69999701
    },
    clipPath: "url(#".concat(o, ")")
  }, /*#__PURE__*/_react.default.createElement("path", {
    style: {
      fill: "url(#".concat(p, ")"),
      stroke: 'none'
    },
    d: "M139.155 147.088c6.855-6.855 17.969-6.855 24.824 0s6.855 17.969 0 24.824l-64.947 64.946c-6.855 6.855-17.969 6.855-24.824 0s-6.855-17.969 0-24.823z"
  }))), /*#__PURE__*/_react.default.createElement("path", {
    style: {
      fill: "url(#".concat(q, ")"),
      stroke: 'none'
    },
    d: "m204.101 236.858-64.947-64.946c-6.854-6.855-6.854-17.969 0-24.824 6.856-6.855 17.97-6.855 24.824 0l64.947 64.947c6.855 6.855 6.855 17.968 0 24.823A17.495 17.495 0 0 1 216.513 242a17.498 17.498 0 0 1-12.412-5.142",
    transform: "matrix(1.33333 0 0 -1.33333 0 425.333)"
  }), /*#__PURE__*/_react.default.createElement("path", {
    style: {
      fill: "url(#".concat(r, ")"),
      stroke: 'none'
    },
    d: "M253.374 223.794v-71.315c0-41.685 33.793-75.479 75.479-75.479 9.695 0 17.553 7.859 17.553 17.554 0 9.694-7.858 17.553-17.553 17.553-22.297 0-40.372 18.075-40.372 40.372v71.315c0 10.055-7.505 18.205-17.554 18.205s-17.553-8.15-17.553-18.205",
    transform: "matrix(1.33333 0 0 -1.33333 0 425.333)"
  }));
}
