// sanity-plugin-mux-input/build/components/SetupButton.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = SetupButton;
var _ui = require("@sanity/ui");
var _pluginIcon = _interopRequireDefault(require("part:@sanity/base/plugin-icon"));
var _react = _interopRequireDefault(require("react"));
var _Input = _interopRequireDefault(require("./Input.css"));
var _SetupDialog = _interopRequireDefault(require("./SetupDialog"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function SetupButton(_ref) {
  var onCancel = _ref.onCancel,
    onSave = _ref.onSave,
    secrets = _ref.secrets,
    onSetup = _ref.onSetup,
    isLoading = _ref.isLoading,
    showSetup = _ref.showSetup,
    needsSetup = _ref.needsSetup;
  var renderSetup = !isLoading && showSetup;
  return /*#__PURE__*/_react.default.createElement("div", {
    className: _Input.default.setupButtonContainer
  }, /*#__PURE__*/_react.default.createElement(_ui.Button, {
    tone: needsSetup ? 'critical' : 'positive',
    mode: "bleed",
    onClick: onSetup,
    icon: _pluginIcon.default,
    padding: 3,
    radius: 3,
    "aria-label": "Set up Mux credentials"
  }), renderSetup && /*#__PURE__*/_react.default.createElement(_SetupDialog.default, {
    secrets: secrets,
    onCancel: onCancel,
    onSave: onSave
  }));
}
