// sanity-plugin-mux-input/build/components/Setup.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _ui = require("@sanity/ui");
var _lodash = require("lodash");
var _default2 = _interopRequireDefault(require("part:@sanity/components/formfields/default"));
var _react = _interopRequireWildcard(require("react"));
var _secrets = require("../actions/secrets");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function asyncGeneratorStep(gen, resolve, reject, _next, _throw, key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { Promise.resolve(value).then(_next, _throw); } }
function _asyncToGenerator(fn) { return function () { var self = this, args = arguments; return new Promise(function (resolve, reject) { var gen = fn.apply(self, args); function _next(value) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "next", value); } function _throw(err) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "throw", err); } _next(undefined); }); }; }
function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
class MuxVideoInputSetup extends _react.Component {
  constructor() {
    var _this$props$secrets$t, _this$props$secrets, _this$props$secrets$s, _this$props$secrets2, _this$props$secrets$e, _this$props$secrets3, _this$props$secrets$s2, _this$props$secrets4, _this$props$secrets$s3, _this$props$secrets5, _this;
    super(...arguments);
    _this = this;
    _defineProperty(this, "tokenInputId", (0, _lodash.uniqueId)('MuxTokenInput'));
    _defineProperty(this, "secretKeyInputId", (0, _lodash.uniqueId)('MuxSecretInput'));
    _defineProperty(this, "enableSignedUrlsInputId", (0, _lodash.uniqueId)('MuxEnableSignedUrlsInput'));
    _defineProperty(this, "state", {
      token: (_this$props$secrets$t = (_this$props$secrets = this.props.secrets) === null || _this$props$secrets === void 0 ? void 0 : _this$props$secrets.token) !== null && _this$props$secrets$t !== void 0 ? _this$props$secrets$t : null,
      secretKey: (_this$props$secrets$s = (_this$props$secrets2 = this.props.secrets) === null || _this$props$secrets2 === void 0 ? void 0 : _this$props$secrets2.secretKey) !== null && _this$props$secrets$s !== void 0 ? _this$props$secrets$s : null,
      enableSignedUrls: (_this$props$secrets$e = (_this$props$secrets3 = this.props.secrets) === null || _this$props$secrets3 === void 0 ? void 0 : _this$props$secrets3.enableSignedUrls) !== null && _this$props$secrets$e !== void 0 ? _this$props$secrets$e : false,
      signingKeyId: (_this$props$secrets$s2 = (_this$props$secrets4 = this.props.secrets) === null || _this$props$secrets4 === void 0 ? void 0 : _this$props$secrets4.signingKeyId) !== null && _this$props$secrets$s2 !== void 0 ? _this$props$secrets$s2 : null,
      signingKeyPrivate: (_this$props$secrets$s3 = (_this$props$secrets5 = this.props.secrets) === null || _this$props$secrets5 === void 0 ? void 0 : _this$props$secrets5.signingKeyPrivate) !== null && _this$props$secrets$s3 !== void 0 ? _this$props$secrets$s3 : null,
      isLoading: false,
      error: null
    });
    _defineProperty(this, "firstField", /*#__PURE__*/_react.default.createRef());
    _defineProperty(this, "handleTokenChanged", event => {
      this.setState({
        token: event.currentTarget.value
      });
    });
    _defineProperty(this, "handleSecretKeyChanged", event => {
      this.setState({
        secretKey: event.currentTarget.value
      });
    });
    _defineProperty(this, "handleEnableSignedUrls", event => this.setState({
      enableSignedUrls: event.currentTarget.checked
    }));
    _defineProperty(this, "handleOnSubmit", event => {
      event.preventDefault();
    });
    _defineProperty(this, "handleSaveToken", /*#__PURE__*/_asyncToGenerator(function* () {
      _this.setState({
        isLoading: true
      });
      var _this$state = _this.state,
        token = _this$state.token,
        secretKey = _this$state.secretKey,
        enableSignedUrls = _this$state.enableSignedUrls;
      var _this$state2 = _this.state,
        signingKeyId = _this$state2.signingKeyId,
        signingKeyPrivate = _this$state2.signingKeyPrivate;
      try {
        yield (0, _secrets.saveSecrets)(token, secretKey, enableSignedUrls, signingKeyId, signingKeyPrivate);
      } catch (err) {
        console.error('Error while trying to save secrets:', err); // eslint-disable-line no-console
        _this.setState({
          isLoading: false,
          error: 'Something went wrong saving the token. See console.error for more info.'
        });
        return;
      }
      if (enableSignedUrls) {
        var hasValidSigningKeys = yield (0, _secrets.haveValidSigningKeys)(signingKeyId, signingKeyPrivate);
        if (!hasValidSigningKeys) {
          try {
            var _yield$createSigningK = yield (0, _secrets.createSigningKeys)(),
              data = _yield$createSigningK.data;
            signingKeyId = data.id;
            signingKeyPrivate = data.private_key;
            yield (0, _secrets.saveSecrets)(token, secretKey, enableSignedUrls, signingKeyId, signingKeyPrivate);
          } catch (err) {
            // eslint-disable-next-line no-console
            console.log('Error while creating and saving signing key:', err === null || err === void 0 ? void 0 : err.message);
            _this.setState({
              error: err === null || err === void 0 ? void 0 : err.message
            });
          }
        }
      }
      _this.setState({
        isLoading: false
      });
      _this.props.onSave({
        token,
        secretKey,
        enableSignedUrls,
        signingKeyId,
        signingKeyPrivate
      });
    }));
  }
  componentDidMount() {
    this.firstField.current.focus();
  }
  render() {
    var _this$state3 = this.state,
      error = _this$state3.error,
      isLoading = _this$state3.isLoading;
    return /*#__PURE__*/_react.default.createElement(_ui.Box, {
      paddingRight: 4,
      paddingLeft: 4,
      paddingBottom: 4,
      paddingTop: 4,
      style: {
        position: 'relative'
      }
    }, /*#__PURE__*/_react.default.createElement("form", {
      onSubmit: this.handleOnSubmit
    }, /*#__PURE__*/_react.default.createElement(_ui.Stack, {
      space: 4
    }, !this.state.token && /*#__PURE__*/_react.default.createElement(_ui.Card, {
      padding: [3, 3, 3],
      radius: 2,
      shadow: 1,
      tone: "primary"
    }, /*#__PURE__*/_react.default.createElement(_ui.Stack, {
      space: 3
    }, /*#__PURE__*/_react.default.createElement(_ui.Text, {
      size: 1
    }, "To set up a new access token, go to your", ' ', /*#__PURE__*/_react.default.createElement("a", {
      href: "https://dashboard.mux.com/settings/access-tokens",
      target: "_blank",
      rel: "noreferrer noopener"
    }, "account on mux.com"), "."), /*#__PURE__*/_react.default.createElement(_ui.Text, {
      size: 1
    }, "The access token needs permissions: ", /*#__PURE__*/_react.default.createElement("strong", null, "Mux Video "), "(Full Access) and ", /*#__PURE__*/_react.default.createElement("strong", null, "Mux Data"), " (Read)", /*#__PURE__*/_react.default.createElement("br", null), "The credentials will be stored safely in a hidden document only available to editors."))), /*#__PURE__*/_react.default.createElement(_default2.default, {
      changeIndicator: false,
      label: "Access Token",
      labelFor: this.tokenInputId,
      level: 0
    }, /*#__PURE__*/_react.default.createElement(_ui.TextInput, {
      id: this.tokenInputId,
      ref: this.firstField,
      onChange: this.handleTokenChanged,
      type: "text",
      value: this.state.token || ''
    })), /*#__PURE__*/_react.default.createElement(_default2.default, {
      changeIndicator: false,
      label: "Secret Key",
      labelFor: this.secretKeyInputId,
      level: 0
    }, /*#__PURE__*/_react.default.createElement(_ui.TextInput, {
      id: this.secretKeyInputId,
      onChange: this.handleSecretKeyChanged,
      type: "text",
      value: this.state.secretKey || ''
    })), /*#__PURE__*/_react.default.createElement(_ui.Stack, {
      space: 4
    }, /*#__PURE__*/_react.default.createElement(_ui.Flex, {
      align: "center"
    }, /*#__PURE__*/_react.default.createElement(_ui.Checkbox, {
      id: this.enableSignedUrlsInputId,
      onChange: this.handleEnableSignedUrls,
      checked: this.state.enableSignedUrls || false,
      style: {
        display: 'block'
      }
    }), /*#__PURE__*/_react.default.createElement(_ui.Box, {
      flex: 1,
      paddingLeft: 3
    }, /*#__PURE__*/_react.default.createElement(_ui.Text, null, /*#__PURE__*/_react.default.createElement("label", {
      htmlFor: this.enableSignedUrlsInputId
    }, "Enable Signed Urls")))), this.state.signingKeyId && this.state.enableSignedUrls ? /*#__PURE__*/_react.default.createElement(_ui.Card, {
      padding: [3, 3, 3],
      radius: 2,
      shadow: 1,
      tone: "caution"
    }, /*#__PURE__*/_react.default.createElement(_ui.Stack, {
      space: 3
    }, /*#__PURE__*/_react.default.createElement(_ui.Text, {
      size: 1
    }, "The signing key ID that Sanity will use is:"), /*#__PURE__*/_react.default.createElement(_ui.Code, {
      size: 1
    }, this.state.signingKeyId), /*#__PURE__*/_react.default.createElement(_ui.Text, {
      size: 1
    }, "This key is only used for previewing content in the Sanity UI.", /*#__PURE__*/_react.default.createElement("br", null), "You should generate a different key to use in your application server."))) : null), /*#__PURE__*/_react.default.createElement(_ui.Inline, {
      space: 2
    }, /*#__PURE__*/_react.default.createElement(_ui.Button, {
      text: "Save",
      loading: isLoading,
      tone: "primary",
      mode: "default",
      onClick: this.handleSaveToken
    }), /*#__PURE__*/_react.default.createElement(_ui.Button, {
      text: "Cancel",
      tone: "primary",
      mode: "bleed",
      onClick: this.props.onCancel
    })), error && /*#__PURE__*/_react.default.createElement(_ui.Card, {
      padding: [3, 3, 3],
      radius: 2,
      shadow: 1,
      tone: "critical"
    }, /*#__PURE__*/_react.default.createElement(_ui.Text, null, error)))));
  }
}
var _default = MuxVideoInputSetup;
exports.default = _default;
