// media-chrome/dist/media-poster.js
import i from"./media-chrome-html-element.js";import{defineCustomElement as a}from"./utils/defineCustomElement.js";import{Document as r}from"./utils/server-safe-globals.js";const t=r.createElement("template");t.innerHTML=`
<style>
  :host {
    position: absolute;
    width: 100%;
    height: 100%;
  }

  #poster {
    position: absolute;
    left: 0;
    top: 0;
    width: 100%;
    height: 100%;

    background-color: #000;
    background-position: 50% 50%;
    background-repeat: no-repeat;
    background-size: contain;
    transition: opacity 0.2s ease;
    opacity: 1;
  }

  #poster.hidden {
    display: none;
    opacity: 0;
  }
</style>
<div id="poster"></div>
`;class o extends i{constructor(){super();var e=this.attachShadow({mode:"open"});this.shadowRoot.appendChild(t.content.cloneNode(!0)),this.addEventListener("click",s=>{this.onClick(s)})}mediaSetCallback(e){e.addEventListener("play",()=>{this.hide()})}onClick(){const e=this.media;e&&e.play()}set src(e){this.shadowRoot.querySelector("#poster").style.backgroundImage=`url(${e})`}get src(){const e=this.shadowRoot.querySelector("#poster").style.backgroundImage;return!e||e==="none"?null:e.substr(4,e.length-5)}show(){this.shadowRoot.querySelector("#poster").className=""}hide(){this.shadowRoot.querySelector("#poster").className="hidden"}}a("media-poster",o);export default o;
