// media-chrome/dist/media-thumbnail-preview-element.js
import m from"./media-chrome-html-element.js";import{Document as l}from"./utils/server-safe-globals.js";import{defineCustomElement as h}from"./utils/defineCustomElement.js";const s=l.createElement("template");s.innerHTML=`
  <style>
    :host {
      box-sizing: border-box;
      background-color: #000;
      width: 284px;
      height: 160px;
      overflow: hidden;
    }

    img {
      position: absolute;
      left: 0;
      top: 0;
    }
  </style>
  <img crossorigin loading="eager" decoding="async" />
`;class r extends m{static get observedAttributes(){return["time"].concat(super.observedAttributes||[])}constructor(){super();const e=this.attachShadow({mode:"open"});this.shadowRoot.appendChild(s.content.cloneNode(!0))}mediaPreviewImageSet(e){this.update()}mediaPreviewCoordsSet(e){this.update()}update(){if(!this.mediaPreviewCoords||!this.mediaPreviewImage)return;const e=this.shadowRoot.querySelector("img"),[a,d,n,u]=this.mediaPreviewCoords,i=this.mediaPreviewImage,t=this.offsetWidth/n,o=()=>{e.style.width=`${t*e.naturalWidth}px`,e.style.height=`${t*e.naturalHeight}px`};e.src!==i&&(e.onload=o,e.src=i,o()),o(),e.style.left=`-${t*a}px`,e.style.top=`-${t*d}px`}}h("media-thumbnail-preview",r);export default r;
